src/Core/Checkout/Document/SystemCheck/DocumentRenderReadinessCheck.php

```diff
@@ -0,0 +1,329 @@
+                        );
+
+                        continue;
+                    }
+
+                    if ($success === null) {
+                        $result[Status::FAILURE->name] = Status::FAILURE;
+
+                        $extra[] = $this->createExtraEntry(
+                            $documentType,
+                            Status::FAILURE->name,
+                            'Rendering failed without exception (no success result).',
+                            $fileType,
+                        );
+
+                        continue;
+                    }
+
+                    $result[Status::OK->name] = Status::OK;
+
+                    $extra[] = $this->createExtraEntry(
+                        $documentType,
+                        Status::OK->name,
+                        'Rendering successful.',
+                        $fileType,
+                    );
+                } catch (\Throwable $e) {
+                    $result[Status::FAILURE->name] = Status::FAILURE;
+
+                    $extra[] = $this->createExtraEntry(
+                        $documentType,
+                        Status::FAILURE->name,
+                        'Rendering failed with exception: ' . $e->getMessage(),
+                        $fileType,
+                    );
+
+                    continue;
+                }
+            }
+        }
+
+        /**
+         * At the moment, we only set OK and FAILURE statuses.
+         * If there is only one status in the $result array, we return that.
+         * If there are multiple statuses, it means that at least one rendering failed.
+         * So ERROR is returned to indicate that not all renderings were successful.
+         */
+        $finalStatus = \count($result) === 1 ? current($result) : Status::ERROR;
+
+        return new Result(
+            $this->name(),
+            $finalStatus,
+            $finalStatus === Status::OK ? 'All documents rendered successfully.' : 'Some or all documents failed to render.',
+            $finalStatus === Status::OK,
+            $extra
+        );
+    }
+
+    protected function allowedSystemCheckExecutionContexts(): array
+    {
+        return SystemCheckExecutionContext::readiness();
+    }
+
+    private function checkPreconditions(): ?Result
+    {
+        // Check 1: Verify orders exist
+        $orderCount = (int) $this->connection->fetchOne('SELECT COUNT(*) FROM `order`');
+
+        if ($orderCount === 0) {
+            return $this->createEmptyResult(
+                $this->name(),
+                'No orders found; document previews are skipped.'
+            );
+        }
+
+        // Check 2: Verify testable documents exist
+        $sql = '
+            SELECT
+                COUNT(*)
+            FROM `document` AS d
+                INNER JOIN `document_type` AS dt ON d.document_type_id = dt.id
+            WHERE
+                dt.technical_name IN (:documentTypes)
+        ';
```

Comment:
Still nitpick: For this query on the other hand it is much more likely to matter, as it contains a join condition as well as a where clause, so it's more likely the DBMS has to actually count each row that matches these conditions.

But you can apply the same trick here:
```sql
            SELECT
                 1
            FROM `document` AS d
                INNER JOIN `document_type` AS dt ON d.document_type_id = dt.id
            WHERE
                dt.technical_name IN (:documentTypes)
            LIMIT 1
```